js/PanoViewer.js

```diff
@@ -0,0 +1,379 @@
+            } else if (insertion[0] === '.') {
+                return document.getElementsByClassName(insertion.slice(1))[0];
+            }
+            // Trying id
+            element = document.getElementById(insertion);
+        }
+        // If found return it, otherwise go with class name
+        return (element)? element : document.getElementsByClassName(insertion)[0];
+    },
+    getPanoramaData: function(url) {
+        const self = this;
+        return self.getImage(url)
+            .then(self.updatePanoData.bind(self));
+    },
+    updatePanoData: function(response) {
+        const formattedGeometrie = {
+            coordinates: [
+                response.geometrie.coordinates[1],
+                response.geometrie.coordinates[0]
+            ],
+            type: response.geometrie.type
+        };
+        const data = {
+            date: new Date(response.timestamp),
+            id: response.pano_id,
+            hotspots: response.adjacent.map(function (item) {
+                return {
+                    id: item.pano_id,
+                    heading: item.heading,
+                    distance: item.distance,
+                    year: item.year
+                };
+            }),
+            location: this.getCenter(formattedGeometrie),
+            image: response.image_sets.cubic
+        };
+        return {
+            image: data.image,
+            heading: response.heading,
+            pitch: response.pitch,
+            hotspots: data.hotspots
+        }
+    },
+    getImage: function (url) {
+        return new Promise(function (resolve, reject) {
+            const request = new XMLHttpRequest();
+            request.open('GET', url);
+            request.onload = function() {
+                if (request.status === 200) {
+                    try {
+                        response = JSON.parse(request.responseText);
+                        resolve(response);
+                    } catch (e) {
+                        reject(Error(e) + request.responseText);
+                    }
+                } else {
+                    reject(Error('API call returned error code:' + request.statusText));
+                }
+            };
+            // Handling total request failure
+            request.onerror = function() {
+                reject(Error('There was a network error.'));
+            };
+            // Send the request
+            request.send();
+        });
+    },
+    addHotspots: function (hotspots, scene) {
+        const self = this;
+        const container = scene.hotspotContainer();
+        try {
+        hotspots.sort(function (hotspotA, hotspotB) {
+                return hotspotB.distance - hotspotA.distance;
+            }).forEach(function (hotspot) {
+                hs = new Hotspot()
+                hs.init(hotspot.id, self.getConfigInfo('CAMERA_HEIGHT'), hotspot.distance, hotspot.heading, hotspot.year);
```

Comment:
Ik wil de config object niet beschikbaar stellen naar de buiten wereld, maar die wel gebruiken in sommige prototype functions. Ik ken geen andere manier om dit te doen.